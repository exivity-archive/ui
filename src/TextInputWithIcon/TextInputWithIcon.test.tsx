import * as React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { MdVerifiedUser } from 'react-icons/md'

import { TextInputWithIcon } from '.'

test('TextInputWithIcon snapshot', () => {
  const textInputWithIcon = renderer.create(
    <TextInputWithIcon value='TextInputWIthIcon' icon={<MdVerifiedUser/>} onChange={jest.fn()} />
  )

  expect(textInputWithIcon).toMatchSnapshot()
})

test('TextInputWithIcon iconLeft snapshot', () => {
  const textInputWithIcon = renderer.create(
    <TextInputWithIcon iconLeft value='TextInputWIthIcon' icon={<MdVerifiedUser/>} onChange={jest.fn()} />
  )

  expect(textInputWithIcon).toMatchSnapshot()
})

test('value gets changed after an on change event', () => {
  const initialValue = 'initialValue'
  const updatedValue = 'changed'

  const onChangeMock = jest.fn(x => x)

  const textInputWithIcon = mount(<TextInputWithIcon icon={<MdVerifiedUser/>} value={initialValue} onChange={onChangeMock} />)

  textInputWithIcon
    .find('input')
    .simulate('change', { target: { value: updatedValue } })

  expect(onChangeMock.mock.results[0].value).toBe(updatedValue)
})
