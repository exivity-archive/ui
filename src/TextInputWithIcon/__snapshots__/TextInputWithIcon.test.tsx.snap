// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputWithIcon snapshot 1`] = `
.c2 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
  font-size: 16px;
  display: block;
  box-sizing: border-box;
  width: 100%;
  padding: calc(0.5em - 1px) 0.5em;
  border-radius: 4px;
  outline: 0;
  border: 0;
  border: 1px solid #eee;
  background-color: #eee;
  --focus-color: 0,168,216;
}

.c2:hover {
  border-bottom: 1px solid rgba(var(--focus-color),0.5);
}

.c2:focus {
  border-bottom: 1px solid rgba(var(--focus-color),1);
}

.c2::-webkit-input-placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c2::-moz-placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c2:-ms-input-placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c2::placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c2[disabled] {
  cursor: not-allowed;
  box-shadow: inset 0 0 999em rgba(128,128,128,0.2);
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: column;
  -webkit-justify-content: column;
  -ms-flex-pack: column;
  justify-content: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  padding-right: 2.2em;
}

.c3 {
  font-size: 16px;
  height: 1em;
  line-height: 1em;
  text-transform: none;
  position: absolute;
  font-size: 20px;
  right: 0.5em;
}

<div
  className="c0"
>
  <input
    className="c1 c2"
    onChange={[Function]}
    type="text"
    value="TextInputWIthIcon"
  />
  <span
    className="c3"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm-2 16l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"
      />
    </svg>
  </span>
</div>
`;
